src/dtclient/kubernetes_settings.go

```diff
@@ -0,0 +1,251 @@
+}
+
+type getSettingsError struct {
+	Code                 int
+	Message              string
+	ConstraintViolations constraintViolations
+}
+
+type constraintViolations []struct {
+	ParameterLocation string
+	Location          string
+	Message           string
+	Path              string
+}
+
+func (dtc *dynatraceClient) CreateKubernetesSetting(name, kubeSystemUUID, scope string) (string, error) {
+	if name == "" {
+		return "", errors.New("no name given")
+	}
+	if kubeSystemUUID == "" {
+		return "", errors.New("no kube-system namespace UUID given")
+	}
+
+	body := []postKubernetesSettingsBody{
+		{
+			SchemaId:      "builtin:cloud.kubernetes",
+			SchemaVersion: "1.0.27",
+			Value: postKubernetesSettings{
+				Enabled:                         true,
+				Label:                           name,
+				ClusterIdEnabled:                true,
+				ClusterId:                       kubeSystemUUID,
+				CloudApplicationPipelineEnabled: true,
+				OpenMetricsPipelineEnabled:      false,
+				EventProcessingActive:           false,
+				FilterEvents:                    false,
+				EventProcessingV2Active:         false,
+			},
+		},
+	}
+
+	if scope != "" {
+		body[0].Scope = scope
+	}
+
+	bodyData, err := json.Marshal(body)
+	if err != nil {
+		return "", err
+	}
+
+	dtEndpoint := fmt.Sprintf("%s/v2/settings/objects?validateOnly=false", dtc.url)
+
+	req, err := http.NewRequest("POST", dtEndpoint, bytes.NewReader(bodyData))
+	if err != nil {
+		return "", fmt.Errorf("error initializing http request: %s", err.Error())
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Api-Token %s", dtc.apiToken))
+
+	res, err := dtc.httpClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error making post request to dynatrace api: %s", err.Error())
+	}
+
+	resData, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return "", fmt.Errorf("error reading response: %w", err)
+	}
+
+	if res.StatusCode != http.StatusOK &&
+		res.StatusCode != http.StatusCreated {
+		return "", handleErrorArrayResponseFromAPI(resData, res.StatusCode)
+	}
+
+	var resDataJson []postSettingsResponse
+	err = json.Unmarshal(resData, &resDataJson)
+	if err != nil {
+		return "", err
+	}
+
+	if len(resDataJson) != 1 {
+		return "", fmt.Errorf("response is not containing exactly one entry %s", resData)
+	}
+
+	return resDataJson[0].ObjectId, nil
+}
+
+func (dtc *dynatraceClient) GetMonitoredEntitiesForKubeSystemUUID(kubeSystemUUID string) ([]MonitoredEntity, error) {
+	if kubeSystemUUID == "" {
+		return nil, errors.New("no kube-system namespace UUID given")
+	}
+
+	dtEndpoint := fmt.Sprintf("%s/v2/entities", dtc.url)
+
+	req, err := http.NewRequest("GET", dtEndpoint, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing http request: %s", err.Error())
+	}
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Api-Token %s", dtc.apiToken))
```

Comment:
this is similar for every request, could be moved to a function with rest method and url as arguments